geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.geode.management.internal.api;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.cache.configuration.CacheConfig;
+import org.apache.geode.cache.configuration.ClusterCacheElement;
+import org.apache.geode.cache.execute.Function;
+import org.apache.geode.cache.execute.ResultCollector;
+import org.apache.geode.distributed.ConfigurationPersistenceService;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.management.internal.cli.CliUtil;
+import org.apache.geode.management.internal.cli.functions.CliFunctionResult;
+import org.apache.geode.management.internal.cli.functions.UpdateCacheFunction;
+import org.apache.geode.management.internal.exceptions.EntityExistsException;
+import org.apache.geode.management.internal.exceptions.NoMembersException;
+
+public class LocatorClusterManagementService implements ClusterManagementService {
+  private static Logger logger = LogService.getLogger();
+  private InternalCache cache;
+  private ConfigurationPersistenceService persistenceService;
+
+  public LocatorClusterManagementService(InternalCache cache,
+      ConfigurationPersistenceService persistenceService) {
+    this.cache = cache;
+    this.persistenceService = persistenceService;
+  }
+
+  @Override
+  public APIResult createCacheElement(ClusterCacheElement config) {
+    APIResult result = new APIResult();
+    CacheConfig cacheConfig;
+    String group = "cluster";
+
+    // assuming config is validated at this point
+    if (persistenceService != null) {
+      cacheConfig = persistenceService.getCacheConfig(group, true);
+      // see if the element already exists in this group's configuration
+      if (config.getExisting(cacheConfig) != null) {
+        throw new EntityExistsException("cache element " + config.getId() + " already exists.");
+      }
+    }
+
+    // execute function on all members
+    Set<DistributedMember> targetedMembers = findMembers(null, null);
+    if (targetedMembers.size() == 0) {
+      throw new NoMembersException();
+    }
+
+    List<CliFunctionResult> functionResults = executeAndGetFunctionResult(
+        new UpdateCacheFunction(), Arrays.asList(config, ClusterCacheElement.Operation.ADD),
+        targetedMembers);
+    functionResults.forEach(fr -> result.addMemberStatus(
+        fr.getMemberIdOrName(), fr.getStatusMessage(), fr.isSuccessful()));
+
+    // persist configuration
+    if (persistenceService != null) {
```

Comment:
No, because the status only indicates success/failure -- in the case of no persistence, it's a third case which the nullability represents -- again, we can use enums to represent that I guess. Will look into it.

Reply:
Used enums to represent this better